src/framework/xr/xr-manager.js

```diff
@@ -696,7 +696,8 @@ class XrManager extends EventHandler {
 
         if (lengthNew > this.views.length) {
             // add new views into list
-            for (let i = 0; i <= (lengthNew - this.views.length); i++) {
+            const viewAddCount = (lengthNew - this.views.length);
```

Comment:
I was looking at recently when you were refactoring around there, and though it'd be best to replace the if above as as well as the loop with `while (lengthNew > this.views.length) {`

and similar for the else / for loop in the else branch.

Reply:
Done. I don't have a way of testing the logic in the else branch so can you double check the code please 😅